packages/react/src/components/RuleBuilder/Rule.jsx

```diff
@@ -276,7 +277,7 @@ const Rule = ({ rule, onAddRule, onRemoveRule, onChange, columns, i18n }) => {
           tooltipPosition="top"
           iconDescription="Remove rule"
           onClick={onRemoveRule(ruleId)}
-          data-testid={`${ruleId}-remove-rule-button`}
+          testId={`${ruleId}-remove-rule-button`}
```

Comment:
Ahhh, ok. Using `data-testid` worked when the button was a Carbon button...

Reply:
Correct, so when I updated the Button component I realized all of these references in snapshots weren't being passed through, so I updated all instances of carbon buttons to be our Button. Hence a lot of the className changes were `iot-button` is added to the list. It also made it more consistent to always reference `testId` instead of both.

Reply:
Nicely done! We shouldn't be using carbon components if we have our own versions